packages/typespec-ts/test/integration/modelPropertyOptional.spec.ts

```diff
@@ -305,6 +305,349 @@ describe("ModelsPropertyOptional Rest Client", () => {
     }
   });
 
+  it("should get all string literal", async () => {
+    try {
+      const result = await client
+        .path("/type/property/optional/string/literal/all")
+        .get();
+      assert.strictEqual(result.status, "200");
+      assert.strictEqual(result.body.property, "hello");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get default string literal", async () => {
+    try {
+      const result = await client
+        .path("/type/property/optional/string/literal/default")
+        .get();
+      assert.strictEqual(result.status, "200");
```

Comment:
assert result detail and this rule would apply to rest of cases.

Reply:
I have updated the test file as you said.